### PR TITLE
[FLINK-26633][elasticsearch] Add recordSend metric to elasticsearch sink

### PR DESCRIPTION
**Pull request by fapaul**
## What is the purpose of the change

 Add recordSend metric to elasticsearch sink

## Brief change log

Increment the recordsSend counter for every request that is generated by the user from a single record.

## Verifying this change

Added a test to verify the counter increases

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-connectors/flink-connector-elasticsearch-base/src/test/java/org/apache/flink/connector/elasticsearch/sink/ElasticsearchWriterITCase.java [hunk 1]
```diff
@@ -204,6 +205,28 @@ void testIncrementByteOutMetric() throws Exception {
...
+        final BulkProcessorConfig bulkProcessorConfig =
+                new BulkProcessorConfig(flushAfterNActions, -1, -1, FlushBackoffType.NONE, 0, 0);
+
+        try (final ElasticsearchWriter<Tuple2<Integer, String>> writer =
+                createWriter(index, false, bulkProcessorConfig)) {
+            final Optional<Counter> recordsSend =
+                    metricListener.getCounter(MetricNames.NUM_RECORDS_SEND);
+            writer.write(Tuple2.of(1, buildMessage(1)), null);
+            writer.write(Tuple2.of(2, buildMessage(2)), null);
+            writer.write(Tuple2.of(2, buildMessage(3)), null);
```
Comment (JingGe) [inline flink-connectors/flink-connector-elasticsearch-base/src/test/java/org/apache/flink/connector/elasticsearch/sink/ElasticsearchWriterITCase.java:221]: Could these three write calls test the three add(...) methods, i.e. DeleteRequest, IndexRequest, UpdateRequest

**Review by JingGe — COMMENTED** [review]
Thanks @fapaul for driving this PR. It looks overall great. I just left one comment. Please let me know your thoughts.

File: flink-connectors/flink-connector-elasticsearch-base/src/test/java/org/apache/flink/connector/elasticsearch/sink/ElasticsearchWriterITCase.java (hunk 1 above)
Comment (fapaul) [reply flink-connectors/flink-connector-elasticsearch-base/src/test/java/org/apache/flink/connector/elasticsearch/sink/ElasticsearchWriterITCase.java]: I updated the test to send also an update and delete request. Please take another look.

**Review by JingGe — APPROVED** [review]
Thanks @fapaul for your effort! LGTM

**Comment by fapaul** [comment]
@flinkbot run azure

**Pull request merged**